src/words/TokTok.jl

```diff
@@ -115,87 +107,109 @@ end
 
 """
     handle_final_periods(::TokenBuffer)
-Handles the following rules using TokenBuffer API:  
-FINAL_PERIOD_1
-FINAL_PERIOD_2
+Handles the following rules from original toktok perl script:
+Don't tokenize period unless it ends the line and that it isn't preceded by another period (FINAL_PERIOD_1)
+Don't tokenize period unless it ends the line(FINAL_PERIOD_2)
 """
 function handle_final_periods(ts::TokenBuffer)
-    flag = length(ts.input)
+    effective_end = length(ts.input)
     # handles FINAL_PERIOD_1 = r"(?<!\.)\.$"
     if length(ts.input) >= 2 && ts.input[end] == "." && ts.input[end-1] != "."
         flush!(ts, ".")
-        flag -= 1
+        effective_end -= 1
+        return effective_end
     end
 
     # handles FINAL_PERIOD_2 = r"(?<!\.)\.\s*(["'’»›”]) *$"
-    if ts.input[end] in string.(Tuple("“”‘’›"))
-        i = flag - 1
-        while ts.input[i] in RSTRIP[1]
-            i -= 1
+    if ts.input[end] in "“”‘’›" || isspace(ts.input[end]) 
+        while effective_end >=1 && isspace(ts.input[effective_end] )
+            effective_end -= 1
         end
-        if ts.input[i] == "."
-            flag = i - 1
-            flush!(ts, ts.input[i:end])
+        
+        if ts.input[effective_end] in "“”‘’›"
+            token_position = effective_end
+            effective_end -= 1
+            
+            while effective_end >=1 && isspace(ts.input[effective_end] )
+                effective_end -= 1
+            end
+            
+            if ts.input[effective_end] == '.'
+                if effective_end >= 2 && ts.input[effective_end - 1] == '.'
+                    return length(ts.input)
+                else
+                    flush!(ts, ".")
+                    flush!(ts, string(ts.input[token_position]))
+                    effective_end -= 1
+                end
+            end
         end
     end
-    return flag
+    return effective_end
 end
 
+# In below functions flush!() is used when some given string needs to be a seperate token
```

Comment:
Stated the use of flush! and push! as asked

Reply:
Excellent